src/ratelimit.ts

```diff
@@ -22,32 +22,32 @@ export const RateLimiter = (
     }
 
     if (
-      opt.store.has(ip) &&
-      timestamp - opt.store.get(ip)!.lastRequestTimestamp > opt.windowMs
+      await opt.store.has(ip) &&
+      timestamp - (await opt.store.get(ip)!).lastRequestTimestamp > opt.windowMs
     ) {
       opt.store.delete(ip);
     }
-    if (!opt.store.has(ip)) {
+    if (!await opt.store.has(ip)) {
       opt.store.set(ip, {
         remaining: opt.max,
         lastRequestTimestamp: timestamp,
       });
     }
 
-    if (opt.store.has(ip) && opt.store.get(ip)!.remaining === 0) {
+    if (await opt.store.has(ip) && (await opt.store.get(ip)!).remaining === 0) {
       opt.onRateLimit(ctx, next, opt);
     } else {
       await next();
       if (opt.headers) {
         ctx.response.headers.set(
           "X-RateLimit-Remaining",
-          opt.store.get(ip)
-            ? opt.store.get(ip)!.remaining.toString()
+          await opt.store.get(ip)
+            ? (await opt.store.get(ip)!).remaining.toString()
             : opt.max.toString(),
         );
       }
       opt.store.set(ip, {
-        remaining: opt.store.get(ip)!.remaining - 1,
+        remaining: (await opt.store.get(ip)!).remaining - 1,
```

Comment:
I would prefer you have any statements which require a callback (`await`) to be their own variable to avoid promise stacking and make for cleaner and understandable code.